controllers/dspipeline_controller.go

```diff
@@ -291,16 +253,187 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, err
 	}
 
+	err, conditions := r.GenerateStatus(ctx, dspa)
+	if err != nil {
+		log.Info(err.Error())
+		return ctrl.Result{}, err
+	}
+	dspa.Status.Conditions = conditions
+
+	// Update Status
+	err = r.Status().Update(ctx, dspa)
+	if err != nil {
+		log.Info(err.Error())
+		return ctrl.Result{}, err
+	}
+
+	r.PublishMetrics(
+		dspa,
+		util.GetConditionByType(config.APIServerReady, conditions),
+		util.GetConditionByType(config.PersistenceAgentReady, conditions),
+		util.GetConditionByType(config.ScheduledWorkflowReady, conditions),
+		util.GetConditionByType(config.CrReady, conditions),
+	)
+
+	return ctrl.Result{}, nil
+}
+
+// isDeploymentInCondition evaluates if condition with "name" is in condition of type "conditionType".
+// this procedure is valid only for conditions with bool status type, for conditions of non bool type
+// results are undefined.
+func (r *DSPAReconciler) handleReadyCondition(
+	ctx context.Context,
+	dspa *dspav1alpha1.DataSciencePipelinesApplication,
+	name string,
+	condition string,
+) (error, metav1.Condition) {
```

Comment:
by convention we should probably reverse the return tuple here (`$returnvalue, $error`)